include/aws/http/request_response.h

```diff
@@ -296,6 +296,60 @@ struct aws_http_request_handler_options {
     aws_http_on_stream_complete_fn *on_complete;
 };
 
+struct aws_http1_stream_chunk;
+
+/**
+ * Signature definition of the function which will be invoked, if provided by the caller, after a chunked transfer
+ *  encoding chunk has been written to an HTTP/1.1 connection.
+ */
+typedef void aws_http1_stream_write_chunk_complete_fn(struct aws_http1_stream_chunk *chunk, void *user_data);
+
+/**
+ * HTTP/1.1 chunk extension for chunked encoding.
+ * Note that the underlying strings are not owned by the byte cursors.
+ */
+struct aws_http1_chunk_extension {
+    struct aws_byte_cursor key;
+    struct aws_byte_cursor value;
+};
+
+/**
+ * Encoding options for an HTTP/1.1 chunked transfer encoding chunk.
+ */
+struct aws_http1_chunk_options {
+    /**
+     * A pointer to an array of chunked extensions. The num_extensions must match the length of the array.
+     */
+    struct aws_http1_chunk_extension *extensions;
+
+    /**
+     * The number of elements defined in the extensions array.
+     */
+    size_t num_extensions;
+
+    /**
+     * Holds the number of extensions already sent over the connection.
+     */
+    size_t sent_extensions;
```

Comment:
rather keep implementation details like `sent_headers` out of this public struct.
Maybe move it to `aws_http1_stream_chunk` since that has other internal details like `node`.

Reply:
gotcha.